### ISSUE TITLE
runtime: openbsd-arm broken using unreleased toolchain for GOROOT_BOOTSTRAP

### ISSUE THREAD

**Issue by bradfitz** [golang/go#14719]
openbsd-arm is broken:

```
##### Building go_bootstrap for host, openbsd/arm.
runtime/internal/sys
fatal error: maxGCMask too small for now

goroutine 1 [running]:
runtime.gothrow(0x54d618, 0x1b)
	/home/gopher/src/go1.4/src/runtime/panic.go:507 +0x8c fp=0x10c5cc78 sp=0x10c5cc6c
runtime.mallocgc(0x1e0000, 0x503220, 0x0, 0x3)
	/home/gopher/src/go1.4/src/runtime/malloc.go:245 +0x59c fp=0x10c5cd00 sp=0x10c5cc78
runtime.newobject(0x503220, 0x6)
	/home/gopher/src/go1.4/src/runtime/malloc.go:387 +0x4c fp=0x10c5cd14 sp=0x10c5cd00
bootstrap/internal/obj.Linknew(0x9ab228, 0xbffc0b63)
	/home/gopher/build/openbsd-arm-b20187196f26/go/src/cmd/internal/obj/sym.go:80 +0x28 fp=0x10c5cd64 sp=0x10c5cd14
bootstrap/compile/internal/gc.Main()
	/home/gopher/build/openbsd-arm-b20187196f26/go/src/cmd/compile/internal/gc/lex.go:109 +0x278 fp=0x10c5cf40 sp=0x10c5cd64
bootstrap/compile/internal/arm.Main()
	/home/gopher/build/openbsd-arm-b20187196f26/go/src/cmd/compile/internal/arm/galign.go:81 +0x61c fp=0x10c5cf54 sp=0x10c5cf40
main.main()
	/home/gopher/build/openbsd-arm-b20187196f26/go/src/cmd/compile/main.go:34 +0x32c fp=0x10c5cfc0 sp=0x10c5cf54
runtime.main()
	/home/gopher/src/go1.4/src/runtime/proc.go:81 +0x1f4 fp=0x10c5cfe4 sp=0x10c5cfc0
runtime.goexit()
	/home/gopher/src/go1.4/src/runtime/asm_arm.s:1312 +0x4 fp=0x10c5cfe4 sp=0x10c5cfe4
go tool dist: FAILED: /home/gopher/build/openbsd-arm-b20187196f26/go/pkg/tool/openbsd_arm/compile -pack -o /tmp/go-tool-dist-154911004/runtime/internal/sys/_go_.a -p runtime/internal/sys /home/gopher/build/openbsd-arm-b20187196f26/go/src/runtime/internal/sys/arch_arm.go /home/gopher/build/openbsd-arm-b20187196f26/go/src/runtime/internal/sys/stubs.go /home/gopher/build/openbsd-arm-b20187196f26/go/src/runtime/internal/sys/sys.go /home/gopher/build/openbsd-arm-b20187196f26/go/src/runtime/internal/sys/zgoarch_arm.go /home/gopher/build/openbsd-arm-b20187196f26/go/src/runtime/internal/sys/zgoos_openbsd.go /home/gopher/build/openbsd-arm-b20187196f26/go/src/runtime/internal/sys/zversion.go: exit status 2
Build complete, duration 4m23.402398663s. Result: error: exit status 2
```

/cc @minux @4a6f656c

**Comment by diegomontoya** [comment]
Also happening on tip with linux/amd64

##### Building go_bootstrap for host, linux/amd64.
runtime/internal/sys
fatal error: maxGCMask too small for now

goroutine 1 [running]:
runtime.throw(0xb8b450, 0x1b)
	/home/zjl/go/src/runtime/panic.go:543 +0x96 fp=0xc208041898 sp=0xc208041880
runtime.heapBitsSetType(0xc2080c6000, 0x2c0000, 0x2bf480, 0xafd1e0)
	/home/zjl/go/src/runtime/mbitmap.go:470 +0x14d fp=0xc208041930 sp=0xc208041898
runtime.mallocgc(0x2c0000, 0xafd1e0, 0x0, 0x1)
	/home/zjl/go/src/runtime/malloc.go:649 +0x631 fp=0xc208041a00 sp=0xc208041930
runtime.newobject(0xafd1e0, 0x6)
	/home/zjl/go/src/runtime/malloc.go:725 +0x49 fp=0xc208041a28 sp=0xc208041a00
bootstrap/internal/obj.Linknew(0x109a820, 0x5)
	/opt/go1.5/src/cmd/internal/obj/sym.go:80 +0x32 fp=0xc208041ac8 sp=0xc208041a28
bootstrap/compile/internal/gc.Main()
	/opt/go1.5/src/cmd/compile/internal/gc/lex.go:109 +0x2b1 fp=0xc208041e30 sp=0xc208041ac8
bootstrap/compile/internal/amd64.Main()
	/opt/go1.5/src/cmd/compile/internal/amd64/galign.go:113 +0x5db fp=0xc208041eb8 sp=0xc208041e30
main.main()
	/opt/go1.5/src/cmd/compile/main.go:32 +0x395 fp=0xc208041f90 sp=0xc208041eb8
runtime.main()
	/home/zjl/go/src/runtime/proc.go:111 +0x2c2 fp=0xc208041fe0 sp=0xc208041f90
runtime.goexit()

**Comment by minux** [comment]
A few days ago I investigated this build failure. Here is my write up:

Starting from rev f388b58540c2cc401dfa79e3d7b6a1846ac9a59f (
https://golang.org/cl/19868)
which introduced a huge type cmd/internal/obj.Link,  bootstrapping with pre
Go 1.5 builds will fail with:

# runtime/internal/sys
fatal error: maxGCMask too small for now
// snip

Solution:
you need to rebuild the bootstrap toolchain to include at least rev
512f75e8dfc7653c873971480c23ba308307b85a (https://golang.org/cl/9888).

One simple solution is to just rebuild the bootstrap toolchain at Go 1.5
release.
If you're unable to build Go 1.5, use the current bootstrap toolchain to
build
rev 45c2e38b37e40c3b6bdc508065edce0643f79110 (the commit immediately
preceding the offending commit) and then use the newly build toolchain as
bootstrap toolchain.

**Comment by mwhudson** [comment]
tip not being buildable with Go 1.4 is a bug, surely?

**Comment by diegomontoya** [comment]
Instead of using an old go1.4  src build (unsure version)  as bootstrap, I download the latest 1.4.3 binary runtime package, used it as goroot_bootstrap and the build error for tip is resolved.

**Comment by minux** [comment]
The message is only present in post Go 1.4 and pre Go 1.5
commits, so this should only affect users of newly introduced
ports that uses an affected commit between Go 1.4 and Go 1.5.

Using either released version of Go 1.4/1.5 should be fine.

**Comment by mwhudson** [comment]
Ah OK.
